modules/beamdyn/src/BeamDyn.f90

```diff
@@ -3237,161 +3303,84 @@ END SUBROUTINE BD_GyroForce
-                                                      !! index 3 = [1,4] (each column of kp_coord)
-   REAL(BDKi),    INTENT(  OUT):: segment_eta(:)      !< ratio of segment length to element length of a beam's member - computed based on Spline basis
-
-   REAL(BDKi)                  :: eta0
-   REAL(BDKi)                  :: eta1
-   REAL(BDKi)                  :: temp_pos0(3)
-   REAL(BDKi)                  :: temp_pos1(3)
-   REAL(BDKi)                  :: sample_step
-   REAL(BDKi)                  :: dist_to_member_start
-   REAL(BDKi)                  :: segment_length(size(kp_coordinate,1)-1) ! segment length using spline basis
-   REAL(BDKi)                  :: member_length(member_total) ! member length using spline basis or FE quadrature
-   INTEGER(IntKi), parameter   :: sample_total = 3
-
-   INTEGER(IntKi)              :: i
-   INTEGER(IntKi)              :: j
-   INTEGER(IntKi)              :: k
-   INTEGER(IntKi)              :: m
-   INTEGER(IntKi)              :: temp_id
-   INTEGER(IntKi)              :: id0
-   INTEGER(IntKi)              :: id1
-
-
-   member_length  = 0.0_BDKi ! initialize to zero
-   segment_length = 0.0_BDKi ! initialize to zero
-
-   temp_id = 0
-   DO i=1,member_total
-       IF(i .EQ. 1) THEN
-           id0 = 1
-           id1 = kp_member(i)
-       ELSE
-           id0 = id1
-           id1 = id0 + kp_member(i) - 1
-       ENDIF
-
-       DO m=1,kp_member(i)-1
-           temp_id = temp_id + 1
-           sample_step = (kp_coordinate(id0+m,3) - kp_coordinate(id0+m-1,3))/(sample_total-1)
-           DO j=1,sample_total-1
-               eta0 = kp_coordinate(temp_id,3) + (j-1)*sample_step
-               eta1 = kp_coordinate(temp_id,3) +     j*sample_step
-               DO k=1,3 ! x-y-z coordinate
-                   temp_pos0(k) = SP_Coef(temp_id,1,k) + SP_Coef(temp_id,2,k)*eta0 + SP_Coef(temp_id,3,k)*eta0**2 + SP_Coef(temp_id,4,k)*eta0**3
-                   temp_pos1(k) = SP_Coef(temp_id,1,k) + SP_Coef(temp_id,2,k)*eta1 + SP_Coef(temp_id,3,k)*eta1**2 + SP_Coef(temp_id,4,k)*eta1**3
-               ENDDO
-               temp_pos1 = temp_pos1 - temp_pos0 ! array of length 3
-               segment_length(temp_id) = segment_length(temp_id) + TwoNorm(temp_pos1)
-           ENDDO
-           member_length(i) = member_length(i) + segment_length(temp_id)
-       ENDDO
-   ENDDO
-
-   ! ratio of segment's length compared to member length
-   temp_id = 0
-   DO i=1,member_total
-       dist_to_member_start = 0.0_BDKi
-       DO j=1,kp_member(i)-1
-           temp_id = temp_id + 1
-           dist_to_member_start = dist_to_member_start + segment_length(temp_id)
-           segment_eta(temp_id) = dist_to_member_start/member_length(i)
-       ENDDO
-   ENDDO
-
-END SUBROUTINE BD_SegmentEta
-
+!SUBROUTINE BD_diffmtc( nodes_per_elem,GLL_nodes,QPtN,nqp,Shp,ShpDer )
```

Comment:
@michaelasprague Was leaving the commented out code intentional ?

Reply:
There isn't any reason to leave the commented `BD_diffmtc` in BeamDyn.f90; it has been relocated to BeamDyn_Subs.f90